glib/src/gstring.rs

```diff
@@ -404,6 +404,48 @@ impl<'a> ToGlibPtr<'a, *mut c_char> for GString {
     }
 }
 
+#[doc(hidden)]
+impl<'a> FromGlibContainer<*const c_char, *const i8> for GString {
+    unsafe fn from_glib_none_num(ptr: *const i8, num: usize) -> Self {
+        if num == 0 {
+            return Self::from("");
+        }
+        let mut bytes = Vec::with_capacity(num);
+
+        for i in 0..num {
+            let b = std::ptr::read(ptr.add(i)) as u8;
+            bytes.push(b);
+        }
```

Comment:
```rust
let slice = std::slice::from_raw_parts(ptr as *const u8, num);
bytes.extend_from_slice(slice);
```

That does the same as your loop but more efficiently and shorter.

Reply:
Ah I somehow got confused by the docs of slice::from_raw_parts where it says that num shouldn't be the number of bytes but rather the number of elements.

Reply:
That's the same for the `FromGlibContainer` trait btw (and for `u8` / `i8` it's all the same anyway).